Wrapping/Generators/Python/itk/support/itkTemplate.py

```diff
@@ -22,110 +22,16 @@
-
-
-def normalizeName(name):
-    """Normalize the class name to remove ambiguity
-
-    This function removes the white spaces in the name, and also
-    remove the pointer declaration "*" (it have no sense in python)"""
-
-    name = name.replace(" ", "")
-    name = name.replace("*", "")
-
-    return name
-
+from ..conf import itkConfig
 
-class TemplateTypeError(TypeError):
-    def __init__(self, template_type, input_type):
-        import itkExtras
-
-        def tuple_to_string_type(t):
-            if type(t) == tuple:
-                return ", ".join(itkExtras.python_type(x) for x in t)
-            else:
-                itkExtras.python_type(t)
-
-        import itk
-
-        # Special case for ITK readers: Add extra information.
-        extra_eg = ""
-        if template_type in [
-            itk.ImageFileReader,
-            itk.ImageSeriesReader,
-            itk.MeshFileReader,
-        ]:
-            extra_eg = """
-
-or
-
-    e.g.: image = itk.imread(my_input_filename, itk.F)
-"""
-
-        python_template_type = itkExtras.python_type(template_type)
-        python_input_type = tuple_to_string_type(input_type)
-        type_list = "\n".join(
-            [itkExtras.python_type(x[0]) for x in template_type.keys()]
-        )
-        eg_type = ", ".join(
-            [itkExtras.python_type(x) for x in list(template_type.keys())[0]]
-        )
-        msg = """{template_type} is not wrapped for input type `{input_type}`.
-
-To limit the size of the package, only a limited number of
-types are available in ITK Python. To print the supported
-types, run the following command in your python environment:
-
-    {template_type}.GetTypes()
-
-Possible solutions:
-* If you are an application user:
-** Convert your input image into a supported format (see below).
-** Contact developer to report the issue.
-* If you are an application developer, force input images to be
-loaded in a supported pixel type.
-
-    e.g.: instance = {template_type}[{eg_type}].New(my_input){extra_eg}
-
-* (Advanced) If you are an application developer, build ITK Python yourself and
-turned to `ON` the corresponding CMake option to wrap the pixel type or image
-dimension you need. When configuring ITK with CMake, you can set
-`ITK_WRAP_${{type}}` (replace ${{type}} with appropriate pixel type such as
-`double`). If you need to support images with 4 or 5 dimensions, you can add
-these dimensions to the list of dimensions in the CMake variable
-`ITK_WRAP_IMAGE_DIMS`.
-
-Supported input types:
-
-{type_list}
-""".format(
-            template_type=python_template_type,
-            input_type=python_input_type,
-            type_list=type_list,
-            eg_type=eg_type,
-            extra_eg=extra_eg,
-        )
-        TypeError.__init__(self, msg)
+from ..support import itkBase
+from ..support.itkExtras import output
+from ..support.itkTypes import itkCType
+import math
 
 
-class itkTemplate(object):
```

Comment:
Why is the class renamed? I don't think the name is better, and it may cause backwards compatibility issues.

Reply:
It was renamed because the name of the module and the name of the class were the same, and the module was included both with full path and by lifting the symbols.  It became impossible to distinguish between functions in the module and static member functions in the class.  There was a bug that I was chasing down that greatly confounded by the ambiguity in the same naming for a different context.

Reply:
PS:  I agree the name is not better by itself, but the fact that it is different from the module is important.  I could also rename the name of the module ... the same potential backward compatibility problem could occur.

Reply:
@thewtex I can change this back, but it does make debugging and maintenance a lot harder.  Please let me know if you think the potential for backward compatibility issues outweighs the ability to debug the code more easily.

Reply:
Yes, having a different module and class name would be nice, but I do think we should avoid the change in this case. Thanks, Hans.

Reply:
Reverted to support the previous naming.    Broke class into two components, one for the static base class, one for the non-static components.